llms/mlx_lm/models/llama.py

```diff
@@ -124,7 +124,7 @@ def __init__(self, args: ModelArgs):
         self.post_attention_layernorm = RMSNorm(args.hidden_size, eps=args.rms_norm_eps)
         self.args = args
 
-    def __call__(
+    def forward(
```

Comment:
@angeloskath maybe you have a better idea for how to checkpoint all the models. What I found to be the simplest is to change `__call__` to `forward` and then set the `forward` to the checkpointed function in the main training file.

Note monkey patching `__call__` [does not work for this](https://stackoverflow.com/questions/38541015/how-to-monkey-patch-a-call-method) as far as I can tell. I didn't know this before but evidently `a.__call__` is actually 

I will make this change for all the models unless you have a better idea.

Reply:
Interesting. First cause I had no idea of the `__call__` weird semantics and secondly cause doing

```python
for i in range(model.layers):
    model.layers[i] = nn.utils.checkpoint(model.layers[i])
```

will obviously break a bunch of other things like accessing the parameters. Maybe we should do proper monkey-patching if `checkpoint` is passed module.

Another option is passing a `checkpoint` argument to the module containing the layers but having a `forward()` seems fine. We are not using `__call__` for anything except as a convention sometimes which almost always has an alternative path (eg `nn.utils.checkpoint` which can also take a function).

Reply:
lol that was the first thing I tried.

Reply:
I thought about the arg but it was slightly more involved to wire it all the way through

Reply:
A proper monkey patch of module is interesting if we can work it